### PR TITLE
[web] Move context menu handling to its own class

### PR DESCRIPTION
**Pull request by mdebbar**
Remove all `contextmenu` responsibilities out of `FlutterViewEmbedder`/`EmbeddingStrategy`, and into its own `ContextMenu` class that's instantiated and managed by the view.

There's one major difference that this PR brings: the `contextmenu` event listener is now attached to the `<flutter-view>` element instead of `window`. Since the entire app is contained within `<flutter-view>`, I expect no issues with this change.

Part of https://github.com/flutter/flutter/issues/134443

### CODE REVIEW

**Review by ditman — APPROVED** [review]
Great change. I need to figure out if the context menu enable/disable is going to be tied to a viewID or not (it seems that it should, otherwise we'd need to loop all existing views to disable/enable the context menu?)

**Comment by mdebbar** [comment]
> I need to figure out if the context menu enable/disable is going to be tied to a viewID or not (it seems that it should, otherwise we'd need to loop all existing views to disable/enable the context menu?)

Yeah, I think it should be tied to a view ID. It was probably an oversight. That's why I actually want to use `implicitView!` instead of `implicitView?`. That way, we will be forced to remember to fix this before making `implicitiView` truly optional.

File: lib/web_ui/lib/src/engine/window.dart [hunk 1]
```diff
@@ -36,8 +37,10 @@ const int kImplicitViewId = 0;
 /// In addition to everything defined in [ui.FlutterView], this class adds
 /// a few web-specific properties.
 abstract interface class EngineFlutterView extends ui.FlutterView {
-  MouseCursor get mouseCursor;
   DomElement get rootElement;
+
+  MouseCursor get mouseCursor;
+  ContextMenu get contextMenu;
```
[the remark this answers is not in the document]
Comment (mdebbar) [reply lib/web_ui/lib/src/engine/window.dart:43]: > How are you sorting these?

I'm not really sorting them. Just grouping them. The bottom group contains instances of helper classes.

I think I'll just do alphabetical order like you suggested :)

**Comment by ditman** [comment]
> Yeah, I think it should be tied to a view ID. It was probably an oversight. That's why I actually want to use implicitView! instead of implicitView?. That way, we will be forced to remember to fix this before making implicitiView truly optional.

Yeah, I've been reading about the "implicit view" and Desktop is going to migrate everybody away from its usage. I think we can do the same in the web (prepping design doc!)

**Pull request merged**